mParticle-Apple-SDK/Data Model/MPConsumerInfo.mm

```diff
@@ -176,39 +176,54 @@ - (NSDictionary *)dictionaryRepresentation {
 @end
 
 #pragma mark - MPConsumerInfo
+@interface MPConsumerInfo ()
+@property (nonatomic, strong, nonnull) dispatch_queue_t queue;
+@end
+
 @implementation MPConsumerInfo
 
 @synthesize cookies = _cookies;
 @synthesize mpId = _mpId;
 @synthesize uniqueIdentifier = _uniqueIdentifier;
 
 - (id)init {
+    // Serial since read/write lock not appropriate given how mpId is created.
+    dispatch_queue_t newQueue = dispatch_queue_create("com.mParticle.MPCookie.queue", DISPATCH_QUEUE_SERIAL);
+    
     self = [super init];
-    if (self) {
+    if (self && newQueue) {
```

Comment:
That's not thread safe.  There is a race condition between checking if the queue is created and creation of the queue.  We could isolate it with a dispatch once however like so.
```
- (dispatch_queue_t)queue {
    static dispatch_queue_t theQueue = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        theQueue = dispatch_queue_create("com.mParticle.MPCookie.queue", DISPATCH_QUEUE_SERIAL);
    });
    return theQueue;
}
```
This is the method recommended by apple to perform thread safe lazy initialization.  The queue's underlying value can never be accessed from outside of the function, and dispatch once ensures that it's always initialized only once before returning the value.